datasets/README.md

```diff
@@ -1,28 +1,135 @@
-This will require the same pip dependencies as `submission_runner.py`.
+### WMT 
+From `algorithmic-efficiency` run:
+```bash
+python3 datasets/dataset_setup.py \
+--data_dir $DATA_DIR \
+--wmt
+```
 
-Some datasets require signing a form before downloading:
 
-FastMRI:
-Fill out form on https://fastmri.med.nyu.edu/ and run this script with the
-links that are emailed to you for "knee_singlecoil_train" and
-"knee_singlecoil_val".
+## FastMRI
+Fill out form on https://fastmri.med.nyu.edu/. After filling out the form 
+you should get an email containing the URLS for "knee_singlecoil_train",
+"knee_singlecoil_val" and "knee_singlecoil_test".  
 
-ImageNet:
-Register on https://image-net.org/ and run this script with the links to the
-ILSVRC2012 train and validation images.
+```bash
+python3 datasets/dataset_setup.py \
+--data_dir $DATA_DIR \
+--fastmri \
+--fastmri_knee_singlecoil_train_url '<knee_singlecoil_train_url>' \
+--fastmri_knee_singlecoil_val_url '<knee_singlecoil_val_url>' \
+--fastmri_knee_singlecoil_test_url '<knee_singlecoil_test_url>'
+```
+
+## ImageNet
+Register on https://image-net.org/ and follow directions to obtain the 
+URLS for the ILSVRC2012 train and validation images.
 
-Note for tfds ImageNet, you may have to increase the max number of files allowed
-open at once using `ulimit -n 8192`.
+Imagenet dataset processsing is resource intensive. To avoid potential
+ResourcExhausted errors increase the maximum number of open file descriptors:
+```bash
+ulimit -n 8192
+```
+
+The imagenet data pipeline differs between the pytorch and jax workloads. 
+Therefore, you will have to specify the framework (pytorch or jax) through theframework flag.
+
+```bash
+python3 datasets/dataset_setup.py \ 
+--data_dir /data \
+--imagenet \
+--temp_dir $DATA_DIR/tmp \  
+--imagenet_train_url <imagenet_train_url> \
+--imagenet_val_url <imagenet_val_url\
+--framework jax
 
-Note that in order to avoid potential accidental deletion, this script does NOT
+```
+
+Note that some functions use subprocess.Popen(..., shell=True), which can be
+dangerous if the user injects code into the --data_dir or --temp_dir flags. We
+do some basic sanitization in main(), but submitters should not let untrusted
+users run this script on their systems.
+
+### Cleanup 
+In order to avoid potential accidental deletion, this script does NOT
+delete any intermediate temporary files (such as zip archives) without a user
+confirmation. Deleting temp files is particularly important for Criteo 1TB, as
+there can be multiple copies of the dataset on disk during preprocessing if
+files are not cleaned up. If you do not want any temp files to be deleted, you
+can pass --interactive_deletion=false and then all files will be downloaded to
+the provided --temp_dir, and the user can manually delete these after
+downloading has finished.
+
+## Criteo1tb
+```bash
+python3 datasets/dataset_setup.py \
+--data_dir $DATA_DIR \
+--temp_dir $DATA_DIR/tmp \
+--criteo1tb 
+```
+
+### Clean up 
```

Comment:
delete?

Reply:
I think the clean up part was irrelevant for imagenet but its relevant for criteo so we'll leave it.